### ISSUE TITLE
关于微调模型的预测输出与训练时的验证指标不一致的问题

### ISSUE THREAD

**Issue by Westpsf** [LucaOne/LucaOneTasks#120]
作者您好：

我使用自己的DNA-protein的pair数据微调了模型，类似于ncRPI的二分类任务，使用的脚本是LucaOneTasks/src/training/downstream_tasks/run_ncRPI_lucaone_lucappi2.sh，仅修改了
`seq_max_length_a=12
seq_max_length_b=400`

在运行了10个epoch后，验证集得到了比较好的指标：
`***** Dev results checkpoint-1289988 *****
Test average loss = 0.012318
Test total loss = 70.941124
acc = 0.998611
avg_loss = 0.012318
confusion_matrix = {'tn': 4366, 'fp': 6, 'fn': 2, 'tp': 1385}
f1 = 0.99712
mcc = 0.996207
mcc2 = 0.996207
pr_auc = 0.999678
prec = 0.995687
recall = 0.998558
roc_auc = 0.999899
sn = 0.998558
sp = 0.998628
total_loss = 70.941124
`

但是当我使用训练完的模型，运行prediction.sh 进行预测时：
`export CUDA_VISIBLE_DEVICES=0
python predict_v1_new.py \
    --input_file ../test/ncRPI/ncRPI_prediction.csv \
    --llm_truncation_seq_length 4096 \
    --model_path .. \
    --save_path ../predicts/ncRPI/ncRPI_prediction_results.csv \
    --dataset_name ncRPI \
    --dataset_type gene_protein \
    --task_type binary_class \
    --task_level_type seq_level \
    --model_type lucappi2 \
    --input_type matrix \
    --input_mode pair \
    --time_str 20250811210843 \
    --print_per_num 1000 \
    --step 1289988 \
    --threshold 0.5 \
    --gpu_id 0
`

出现了probs均非常小，因此分类全为0的问题，我列出了部分最后两列的预测输出：
5.930146929067348e-10,0
5.2306869946505685e-08,0
1.0148357352370082e-26,0

即便我使用验证集重新预测也得到了一样的全为0的结果，但是训练时又是良好的指标。
我并不是非常擅长代码，我尝试了阅读源码，但没有什么头绪，我猜测可能是我的使用方式不对，如果能得到您的帮助，我会非常感谢！

**Comment by Westpsf** [comment]
好的谢谢，我明白了，看起来这就是我的问题所在。
我会调整数据然后再试一下，谢谢您的指导！

**Comment by Westpsf** [comment]
感谢指正，问题已解决

**Comment by Westpsf** [comment]
作者您好，我遇到了一个新的问题，预测输出的probs完全取决于seq_a，似乎没有用到任何seq_b的信息，只要seq_a一样，预测的probs的值就完全一样。我更改过负样本生成策略（我尝试了突变seq_a，以及让seq_a，seq_b错配这两种方式），但仍然有一样的问题。训练过程和上面一致，并已经按照您的指正修改了之前的问题

<img width="2155" height="446" alt="Image" src="https://github.com/user-attachments/assets/7235bafa-9740-4215-a193-73f31871c0f6" />

**Comment by Westpsf** [comment]
我查看了预测后的embedding文件，没有问题，seq_a因为序列一样所以embedding都一样，seq_b序列不一样embedding就都不一样
这是我的log.txt文件，

[logs (3).txt](https://github.com/user-attachments/files/22038983/logs.3.txt)

**Comment by Westpsf** [comment]
感谢作者指导，我会按照您的建议再测试一下
